bricks/flutter_clean_base/__brick__/{{project_name.snakeCase()}}/lib/core/arch/widget/common/theme_switcher.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/material.dart';
+
+class ThemeModeSwitcher extends StatefulWidget {
+  final Widget Function(BuildContext, ThemeMode, Widget?) builder;
+  final ThemeMode initialThemeMode;
+
+  const ThemeModeSwitcher({
+    required this.builder,
+    this.initialThemeMode = ThemeMode.system,
+    super.key,
+  });
+
+  @override
+  State<ThemeModeSwitcher> createState() => _ThemeModeSwitcherState();
+}
+
+class _ThemeModeSwitcherState extends State<ThemeModeSwitcher> {
+  late final ValueNotifier<ThemeMode> themeNotifier;
+
+  @override
+  void initState() {
+    super.initState();
+
+    themeNotifier = ValueNotifier(widget.initialThemeMode);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ThemeModeNotifier(
+      notifier: themeNotifier,
+      child: ValueListenableBuilder<ThemeMode>(
+        valueListenable: themeNotifier,
+        builder: widget.builder,
+      ),
+    );
+  }
+}
+
+class ThemeModeNotifier extends InheritedWidget {
+  final ValueNotifier<ThemeMode> notifier;
+
+  const ThemeModeNotifier({
+    required this.notifier,
+    required super.child,
+    super.key,
+  });
+
+  void changeTheme(ThemeMode themeMode) {
+    if (notifier.value != themeMode) {
```

Comment:
by default, the ValueNotifier setter has this logic

![image](https://github.com/user-attachments/assets/df6416c8-d7e8-456c-a0bb-5941582614e9)